apps/system/js/pin_page_system_dialog.js

```diff
@@ -166,6 +167,17 @@
     siteBadge.icon = data.icon;
     this.siteName.textContent = data.name;
 
+    Service.request('Places:isPinned', data.url)
+      .then((isPinned) => {
+        if (isPinned) {
```

Comment:
What about:

```javascript
var action = isPinned ? 'unpin' : 'pin';
this.pinPageButton.setAttribute('data-l10n-id', 'pinning-' + action + '-page');
this.pinPageButton.dataset.action = action + '-page';
```
I think is more readable, but feel free to leave as it is if you prefer